### PR TITLE
Add support for drawing legend from JSON

### PR DESCRIPTION
**Pull request by kdeloach**
Rewrites the legend control so we have more control over how everything
is rendered. Also added support for different ways to render legend
items (single, multiple, and range of values) based on the wireframes.

Connects #482

### CODE REVIEW

**Comment by kdeloach** [comment]
There may be some styling issues remaining.

To test the "scale" legend items you can select the first layer called "Bathymetry".
image

**Comment by kdeloach** [comment]
This logic may be something that we could use as a reference for the new layer selector https://github.com/CoastalResilienceNetwork/GeositeFramework/pull/488/files#diff-39eb965dbc0c1ad41b4e12dbbc16e00bR209. 

It isn't evident from this PR, but we *could* support lazily adding items to the legend now (before the layer is finished loading). I didn't add a "Loading..." UI for aesthetic purposes. 

I can elaborate on this in-person.

File: src/GeositeFramework/Views/Home/Index.cshtml [hunk 1]
```diff
@@ -177,15 +177,54 @@
...
+            <% }) %>
+        </div>
+    </script>
+
+    <script type="text/template" id="template-legend-item-scale">
+        <div class="legend-layer scale">
+            <div class="legend-title"><%- layerName %></div>
+            <% _.each(legend, function(item) { %>
+                <div class="item">
+                    <img src="data:<%= item.contentType %>;base64,<%= item.imageData %>"
```
Comment (mmcfarland) [inline src/GeositeFramework/Views/Home/Index.cshtml]: This isn't in the wireframes, but I think we need to add the text label/value for the first and last values in the scale at the beginning/end of the series.

**Comment by mmcfarland** [comment]
This looks great - can you target `develop` for the PR?  We still haven't made the changes to build that lets us make it the default.

**Comment by kdeloach** [comment]
Created a new PR targeting develop.

File: src/GeositeFramework/Views/Home/Index.cshtml (hunk 1 above)
Comment (kdeloach) [inline src/GeositeFramework/Views/Home/Index.cshtml]: Oh yeah, that makes sense.

**Pull request closed without merging**